src/apps/backend/modules/account/internal/account-reader.ts

```diff
@@ -53,15 +55,50 @@ export default class AccountReader {
     return AccountUtil.convertAccountDBToAccount(accDb);
   }
 
+  public static async getAccountByContactNumber(
+    contactNumber: ContactNumber,
+  ): Promise<Account> {
```

Comment:
Since this function can return null, return type should be `Nullable<Account>`. Also, rename function name to `getAccountByContactNumberOptional` to reflect that this function can return no account